File: ProblemSet1/cash/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

int main(void)
{
    //variables
    
    float giris; //input
    
    int quarter = 25; // quarter value 
    int quartercounter; // counter for how many need quarter
    int dime = 10; // dime value
    int dimecounter; // counter for how many need dime
    int nickel = 5; // nickel value
    int nickelcounter; // counter for how many need nickel
    int penny = 1; // penny value
    int pennycounter; // counter for how many need penny
    
    int total; // counter for total coins
    
    do
    {
        giris = get_float("Change Owed: ");
    }
    while (giris < 0);
    
    //converter from dollars to cents
    
    int cents = round(giris * 100);
    
    
    //total divided by coin
    
    quartercounter = floor(cents / quarter);
    
    //turn remainder into the new total
    
    cents = cents % quarter;
    
    //total divided by coin
    
    dimecounter = floor(cents / dime);
    
    //turn remainder into the new total
    
    cents = cents % dime;
    
    //total divided by coin
    
    nickelcounter = floor(cents / nickel);
    
    //turn remainder into the new total
    
    cents = cents % nickel;
    
    //new total
    pennycounter = cents;
    
    // total coins
    total = quartercounter + dimecounter + nickelcounter + pennycounter;
    
    // printing total coins
    printf("%i\n", total);
    
}

File: ProblemSet6/Cash-Credit/cash.py
#libraries
import cs50
import math

# variables
quarter = 25 
quartercounter = 0 
dime = 10
dimecounter = 0
nickel = 5
nickelcounter = 0
penny = 1
pennycounter = 0
total = 0

#taking input from user
giris = cs50.get_float("Change owed: ")
# checking input is valid
while not(giris > 0):
    giris = cs50.get_float("Change owed: ")

#cents to coins
cents = round(giris * 100)
#calculations
quartercounter = math.floor(cents / quarter)
cents = cents % quarter

dimecounter = math.floor(cents / dime)
cents = cents % dime

nickelcounter = math.floor(cents / nickel)
cents = cents % nickel

pennycounter = cents

#calculating total of counters
total = quartercounter + dimecounter + nickelcounter + pennycounter
#printing total
print(total)